Sources/TuistGenerator/Utils/NamespaceGenerator.swift

```diff
@@ -0,0 +1,84 @@
+import PathKit
+import StencilSwiftKit
+import SwiftGenKit
+import TSCBasic
+import TuistSupport
+
+enum NamespaceType {
+    case assets
+    case strings
+
+    var templateFileName: String {
+        switch self {
+        case .assets:
+            return "xcassets.stencil"
+        case .strings:
+            return "strings.stencil"
+        }
+    }
+}
+
+private extension NamespaceType {
+    func parser() throws -> Parser {
+        switch self {
+        case .assets:
+            return try AssetsCatalog.Parser()
+        case .strings:
+            return try Strings.Parser()
+        }
+    }
+}
+
+protocol NamespaceGenerating {
+    func render(_ namespaceType: NamespaceType, paths: [AbsolutePath]) throws -> [(name: String, contents: String)]
+    func generateNamespaceScript() -> String
+}
+
+final class NamespaceGenerator: NamespaceGenerating {
+    private let resourcesNamespaceTemplatesLocator: ResourcesNamespaceTemplatesLocating
+
+    init(
+        resourcesNamespaceTemplatesLocator: ResourcesNamespaceTemplatesLocating = ResourcesNamespaceTemplatesLocator()
+    ) {
+        self.resourcesNamespaceTemplatesLocator = resourcesNamespaceTemplatesLocator
+    }
+
+    func render(_ namespaceType: NamespaceType, paths: [AbsolutePath]) throws -> [(name: String, contents: String)] {
+        let templatePath = try resourcesNamespaceTemplatesLocator.locateTemplate(for: namespaceType)
+        let template = StencilSwiftTemplate(
+            templateString: try FileHandler.shared.readTextFile(templatePath),
+            environment: stencilSwiftEnvironment()
+        )
+
+        return try paths.map { path in
+            let parser = try namespaceType.parser()
+            try parser.parse(path: Path(path.pathString), relativeTo: Path(""))
+            var context = parser.stencilContext()
+            context = try StencilContext.enrich(context: context, parameters: ["publicAccess": true])
+            return (path.basenameWithoutExt, try template.render(context))
+        }
+    }
+
+    func generateNamespaceScript() -> String {
+        #if DEBUG
+            // Used only for debug purposes to find currently-built tuist
+            // `bundlePath` points to .build/debug/tuist
+            let tuistCommand = AbsolutePath(#file.replacingOccurrences(of: "file://", with: ""))
+                .removingLastComponent()
+                .removingLastComponent()
+                .removingLastComponent()
+                .removingLastComponent()
+                .appending(components: ".build", "debug", "tuist")
+                .pathString
```

Comment:
this may not work debugging in Xcode :/

A slightly more generic way of doing this is to adopt something similar to what we did in `tuist edit`

```swift
       // To be sure that we are using the same binary of Tuist that invoked `edit`
        let tuistPath = AbsolutePath(TuistCommand.processArguments()!.first!)
```

https://github.com/tuist/tuist/blob/master/Sources/TuistKit/ProjectEditor/ProjectEditor.swift#L111

Reply:
this does work in Xcode, but the proposed solution seems nicer, so let me rewrite that 👍

Reply:
hmm, so I don't know if I am able to get to `processArguments()` when I can't touch `TuistCommand` (since it is in `TuistKit`, but I am working in `TuistGenerator`). Any ideas how to get around that, @kwridan ?

Reply:
ah good point I missed this.

I generating the script still needed? if so, perhaps the script generator can be a separate component that lives in `TuistKit` which does have access to `TuistCommand` if needed - this aligns with keeping `TuistGenerator` tuist command agnostic.

Reply:
No, I have dumped the script, at least for now. But this is needed to get `stencil` templates with which we then synthesize the resource interfaces - in a similar fashion as it's done in `TemplatesDirectoryLocator`

Reply:
Ah got it,

A few options I could think of:

- We could move the `SynthesizedResourceInterfaceTemplatesLocator` implementation to `TuistKit` (protocol can remain in `TuistGenerator`) that way the locator could access any tuist command / specific logic to locate the templates

```swift
SynthesizedResourceInterfaceProjectMapper(templatesLocator: SynthesizedResourceInterfaceTemplatesLocator())
```

- Pass a path of the templates directory in the initialiser of the mapper which can be propagate to the necessary components.

```swift
let templatesDirectoryPath = // ...
SynthesizedResourceInterfaceProjectMapper(templateDirectory: templatesDirectoryPath)
```

Possibly the first option is slightly nicer / more explicit as clients of `TuistGenerator` can provide their own implementation and has less room for error, vs the second has an implicit requirement on the structure of the templates directory. 

Does that work?

Reply:
Another option I thought of:

If those templates don’t need be independent files outside of Tuist or user use editable, they could possibly be included in the source code directly as lengthy multi-line strings - it’s not very elegant but could be an option that doesn’t involve additional bundling within the install logic / locating templates etc... _(could be separate source files to ease maintenance)_.

Reply:
Thanks for giving me so many options to choose from! I think I am inclining the most to change the templates to be multi-line strings as it currently brings no real benefits and it incurs some added maintenance since it needs to be handled in the `Installer`.